arbnode/sequencer.go

```diff
@@ -339,7 +367,22 @@ func (s *Sequencer) postTxFilter(header *types.Header, _ *arbosState.ArbosState,
 	if result.Err != nil && result.UsedGas > dataGas && result.UsedGas-dataGas <= s.config().MaxRevertGasReject {
 		return arbitrum.NewRevertReason(result)
 	}
-	s.nonceCache.Update(header, sender, tx.Nonce()+1)
+	newNonce := tx.Nonce() + 1
+	s.nonceCache.Update(header, sender, newNonce)
+	newAddrAndNonce := addressAndNonce{sender, newNonce}
+	nonceFailure, haveNonceFailure := s.nonceFailures.Get(newAddrAndNonce)
+	if haveNonceFailure {
+		nonceFailure.revived = true // prevent the expiry hook from taking effect
```

Comment:
this is not thread safe. Need either some lock or atomic variable to be set-and-replaced both here and in onNonceEvict

Reply:
I think this is fine because only this thread should ever touch the nonceFailures